evm_arithmetization/src/all_stark.rs

```diff
@@ -106,9 +106,24 @@ pub const NUM_TABLES: usize = if cfg!(feature = "cdk_erigon") {
     Table::MemAfter as usize + 1
 };
 
-/// Indices of Keccak Tables
-pub const KECCAK_TABLES_INDICES: [usize; 2] =
-    [Table::Keccak as usize, Table::KeccakSponge as usize];
+/// Indices of optional Tables
+#[cfg(not(feature = "cdk_erigon"))]
+pub const OPTIONAL_TABLE_INDICES: [usize; 5] = [
+    Table::BytePacking as usize,
+    Table::Keccak as usize,
+    Table::KeccakSponge as usize,
+    Table::Logic as usize,
+    Table::MemAfter as usize,
+];
```

Comment:
Curious about how this is handled on a global aspect: why do we need to even specify the optional tables? The CTLs emphasize which table connects to which, which would then fail if a malicious prover was to _hide_ a given table (claiming that for instance there is no Arithmetic Operations). So couldn't we simplify the overall approach?

Reply:
or is that because of the constraining of `ctl_zs_first` below?

Reply:
We need them in the verifier and the root circuit. We don’t want to add conditional verification for all tables due to the additional overhead.

Reply:
Hmm but we could have instead a check against hardcoded empty caps no? That would serve as our current `skip_table` `BoolTarget` for the ctl check (not saying it's better, this would actually add a few gates, but it's only slightly more expensive than the current, not heavy like conditional verification)

Reply:
I mean https://github.com/0xPolygonZero/zk_evm/blob/ce0ab8c861cb220d4008ac3d6b1844e76286f433/evm_arithmetization/src/fixed_recursive_verifier.rs#L1060

Reply:
Ah ok, nevermind then!